From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Josep del Rio <joseprio@google.com>
Date: Wed, 24 Aug 2022 11:00:59 +0000
Subject: [PATCH 048/226] Remove unnecessary HashMap instantiation

Was investigating something else, but noticed something that could be a tiny bit better; return an empty map from `Collections.emptyMap` instead of using a real `HashMap` to send an empty one.

Test: Builds successfully

Change-Id: Id11d39d3ea90e15bd4cb90031d6242d8e1f98819
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 telephony/java/android/telephony/TelephonyManager.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/telephony/java/android/telephony/TelephonyManager.java b/telephony/java/android/telephony/TelephonyManager.java
index b8087c3b02..1b51e3adcb 100644
--- a/telephony/java/android/telephony/TelephonyManager.java
+++ b/telephony/java/android/telephony/TelephonyManager.java
@@ -15250,7 +15250,6 @@ public class TelephonyManager {
         PackageManager.FEATURE_TELEPHONY_MESSAGING
     })
     public Map<Integer, List<EmergencyNumber>> getEmergencyNumberList() {
-        Map<Integer, List<EmergencyNumber>> emergencyNumberList = new HashMap<>();
         try {
             ITelephony telephony = getITelephony();
             if (telephony != null) {
@@ -15263,7 +15262,7 @@ public class TelephonyManager {
             Log.e(TAG, "getEmergencyNumberList RemoteException", ex);
             ex.rethrowAsRuntimeException();
         }
-        return emergencyNumberList;
+        return Collections.emptyMap();
     }
 
     /**
